test/unit/app/services/asset_manager/attachment_updater_test.rb

```diff
@@ -178,7 +178,7 @@ class AssetManager::AttachmentUpdaterTest < ActiveSupport::TestCase
 
     describe "redirect url updates" do
       let(:unpublished_edition) { FactoryBot.create(:unpublished_edition) }
-      let(:redirect_url) { unpublished_edition.public_url }
+      let(:redirect_url) { unpublished_edition.unpublishing.document_url }
```

Comment:
This fix was required when we tweaked the edition factories not to use an after create hook to add the unpublishing. The test was actually incorrect looking at the underlying behaviour, but because the factory was forcibly setting up the unpublishing the values of the edition's public url and the unpublishing document_url were coincidentally the same.